### PR TITLE
Force download file if mime type is text/plain

### PR DESCRIPTION
**Pull request by tonyyb**
Prevent browser to display file content by replacing "text/plain" mime type by "application/force-download"

### CODE REVIEW

File: framework/classes/tools/file.php [hunk 1]
```diff
@@ -143,6 +143,9 @@ public static function send($file, $mime = null, $exit = true)
             } else {
                 $info = \File::file_info($file);
                 empty($mime) or $info['mimetype'] = $mime;
+                if ($info['mimetype'] == 'text/plain') {
```
Comment (jguyomard) [inline framework/classes/tools/file.php]: `===` instead of `==` ?

**Comment by Meroje** [comment]
Isn't it better to properly mark it as a file attachment instead of using the non-registered mimetype ?

edit: I don't think we always want the browser to save the file (displayable or not), in your case maybe you should use the $mime param to force the download ?

**Comment by DOMEYD** [comment]
:+1:

**Comment by tonyyb** [comment]
Now use Content-Disposition: attachment

**Comment by shaoshiva** [comment]
This change has a lot of side effects, with this PR all the files will be forced to download, including images, videos...

I think a better approach would be to set a list of mime types (and/or extensions ?), in a config file, that will be forced to download.

**Comment by Meroje** [comment]
A new method would be best, but I also don't see anything preventing you to set the Content-disposition before calling Tools_File::send()

**Comment by shaoshiva** [comment]
@Meroje Yes, and for an attachment it can be done in the "404.attachmentFound" event.

**Pull request closed without merging**